description: prevent clang from printing unknown-warning warning. Also remove --rtlib=libgcc and -ffile-compilation-dir=. which won't be available until we have clang 13.
author: Wengling Chen <feiyu2817@gmail.com>

---
 build/config/android/BUILD.gn  |    2 --
 build/config/compiler/BUILD.gn |    5 ++++-
 2 files changed, 4 insertions(+), 3 deletions(-)

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1234,7 +1234,7 @@ config("compiler_deterministic") {
     } else {
       # -ffile-compilation-dir is an alias for both -fdebug-compilation-dir=
       # and -fcoverage-compilation-dir=.
-      cflags += [ "-ffile-compilation-dir=." ]
+      cflags += [ "-fdebug-compilation-dir=." ]
     }
     if (!is_win) {
       # We don't use clang -cc1as on Windows (yet? https://crbug.com/762167)
@@ -1616,6 +1616,9 @@ config("default_warnings") {
         cflags += [
           "-Wenum-compare-conditional",
 
+          # Ignore unknown warning warnings from clang
+          "-Wno-unknown-warning-option",
+
           # An ABI compat warning we don't care about, https://crbug.com/1102157
           # TODO(thakis): Push this to the (few) targets that need it,
           # instead of having a global flag.
--- a/build/config/android/BUILD.gn
+++ b/build/config/android/BUILD.gn
@@ -52,11 +52,9 @@ config("compiler") {
   ]
 
   # TODO(crbug.com/1184398): Move to compiler-rt when we are ready.
-  ldflags += [ "--rtlib=libgcc" ]
   if (current_cpu == "arm64") {
     # For outline atomics on AArch64 (can't pass this unconditionally
     # due to unused flag warning on other targets).
-    cflags += [ "--rtlib=libgcc" ]
     if (arm_control_flow_integrity == "standard") {
       cflags += [ "-mbranch-protection=standard" ]
     }
